full/Dockerfile

```diff
@@ -227,6 +227,32 @@ RUN curl -o /tmp/dive.deb -fsSL https://github.com/wagoodman/dive/releases/downl
     && apt install /tmp/dive.deb \
     && rm /tmp/dive.deb
 
+### Cloud Infrastructure CLIs ###
+
+LABEL dazzle/layer=tool-cloud-infra-cli
+LABEL dazzle/test=tests/tool-cloud-infra-cli.yaml
+USER root
+
+# Amazon Web Services
```

Comment:
Consider having each on their own layer as they can change independently of each other?

Also version pinning might be worthwhile as I've been bitten by updates that break existing usages of aws-cli for e.g.

Reply:
❓ Do we need to/is it worth the complexity?

💭 Usually would agree with pinning but as these tools interface with 3rd party vendors whos API's change on a whim I think it's best to run unpinned. If someone wants to pin then they should create their own Dockerimage (which we highly recommend) instead of consuming our defaults.

Reply:
> Do we need to/is it worth the complexity?

Pros and cons (as always), hence the *might be*.

> 3rd party vendors whos API's change on a whim...
>  consuming our defaults.

It is not a particularly stable default i.e. it is whatever was current for that particular vendor when the image was last updated for a likely unrelated reason.

I guess that is acceptable for the generic 'workhorse' image - *a* version of that tool is available, we just don't guarantee which. I don't know the use cases driving the adoption of this in workspace-full as opposed to say workspace-cloud-infra for e.g.

Depending on those use use cases, is it worth considering a check for major version number for each?
e.g. in the tests adding something like:

> - stdout.indexOf("aws-cli/2.") != -1

That way we can say our defaults are 'v2.x of AWS cli' which is somewhat more predictable and perhaps more importantly, documentable.